cdap-examples/WikipediaPipeline/src/main/scala/co/cask/cdap/examples/wikipedia/ScalaSparkKMeans.scala

```diff
@@ -35,9 +35,12 @@ class ScalaSparkKMeans extends SparkMain {
     val maxIterations = arguments.get("max.iterations").map(_.toInt).getOrElse(10)
     val k = arguments.get("num.topics").map(_.toInt).getOrElse(10)
 
+    val dataNamespace = sec.getRuntimeArguments.get(WikipediaPipelineApp.NAMESPACE_ARG)
+
     // Pre-process data for LDA
     val (corpus, vocabArray, _) = ClusteringUtils.preProcess(
-      sc.fromDataset(WikipediaPipelineApp.NORMALIZED_WIKIPEDIA_DATASET), arguments)
+      if (dataNamespace != null) sc.fromDataset(dataNamespace, WikipediaPipelineApp.NORMALIZED_WIKIPEDIA_DATASET)
```

Comment:
Why doing if else in a method call ? Find the RDD before the method call and pass the RDD variable. Also use ternary operator